bin/import_xml2obd.py

```diff
@@ -236,16 +302,38 @@ def upload_resources(remote_site, api_key, jsonfile, resource_directory):
             continue
         res = rec['resources'][0]
         source = resource_directory + '/' + res['url'].split('/')[-1]
-        with open(source) as f:
-            try:
-                rc = site.action.resource_patch(
-                    id=res['id'],
-                    upload=(os.path.basename(source), f))
-            except ckanapi.errors.CKANAPIError:
-                traceback.print_exc()
-                print(rec)
-                print("failed upload " + os.path.basename(source))
-                continue
+        fname='/'.join(['resources', res['id'], res['url'].split('/')[-1]])
+
+        obj = dest.get_obj(fname.lower())
+        skip = False
+        srcmd5 = md5str(source)
+        if obj:
+            objmd5 = base64md5str(obj.properties.content_settings.content_md5)
+            if objmd5 == srcmd5:
+               skip = True
+            else:
+                print (objmd5, srcmd5)
+        if target_pkg:
+            for res in target_pkg['resources']:
+                rmd5 = res.get('hash', None)
+                if rmd5 and srcmd5 in rmd5:
+                    skip = True
+                    break
+        if skip:
+            print('\tsame remote file exists ' + os.path.basename(source))
+        else:
+            with open(source) as f:
+                try:
+                    rc = site.action.resource_patch(
+                        id=res['id'],
+                        hash='md5-%s'%srcmd5,
```

Comment:
use the base64-encoded hash here, I'd like to follow the SRI standard for hashing